lib/fluent/plugin/out_http.rb

```diff
@@ -55,6 +55,8 @@ class RetryableResponse < StandardError; end
     config_param :headers, :hash, default: nil
     desc 'Additional placeholder based headers for HTTP request'
     config_param :headers_from_placeholders, :hash, default: nil
+    desc 'Compress HTTP request body'
+    config_param :compress, :enum, list: [:none, :gzip], default: :none
```

Comment:
It would be better to align with `out_forward` and `out_file`

```suggestion
    config_param :compress, :enum, list: [:text, :gzip], default: :text
```